cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
Sorry, I'm completely lost, maybe it would be better to discuss this on the mailing list from the beginning as by now I have no idea what problem are we trying to solve any longer. All I can say is that:

1. `-Wno-pedantic-ms-format` suppresses the warning that is generated otherwise with `-pedantic` when using 4.8 and 4.9, so it's not clear at all why shouldn't it be used with 4.7+.
1. We already have `CHECK_GCC` macro so if we really need to check for patch level component we should just add it to the existing macro.